tests/api_connexion/endpoints/test_task_instance_endpoint.py

```diff
@@ -1983,15 +1983,15 @@ def test_should_call_mocked_api(self, mock_set_task_instance_state, session):
         self.create_task_instances(session)
 
         NEW_STATE = "failed"
-        mock_set_task_instance_state.return_value = session.get(
-            TaskInstance,
-            {
-                "task_id": "print_the_context",
-                "dag_id": "example_python_operator",
-                "run_id": "TEST_DAG_RUN_ID",
-                "map_index": -1,
-            },
-        )
+        mock_set_task_instance_state.return_value = session.scalars(
+            select(TaskInstance).where(
+                TaskInstance.dag_id == "example_python_operator",
+                TaskInstance.task_id == "print_the_context",
+                TaskInstance.run_id == "TEST_DAG_RUN_ID",
+                TaskInstance.map_index == -1,
+            )
+        ).one_or_none()
+
```

Comment:
The reason for this and other changes in this file is explained in [**Issues identified Point (2)**](https://github.com/apache/airflow/pull/43243#issuecomment-2439774584)